bundles/org.openhab.binding.magentatv/src/main/java/org/openhab/binding/magentatv/internal/handler/MagentaTVControl.java

```diff
@@ -0,0 +1,582 @@
+        }
+
+        logger.debug("Pairing initiated for device {0} (deviceID={1}).", thingConfig.getFriendlyName(),
+                thingConfig.getTerminalID());
+        return true;
+    }
+
+    /**
+     * Calculates the verifificationCode to complete pairing. This will be triggered
+     * as a result after receiving the pairing code provided by the MR. The
+     * verification code is the MD5 hash of <Pairing Code><Terminal-ID><User ID>
+     *
+     * @param pairingCode Pairing code received from the MR
+     * @return true: a new code has been generated, false: the code matches a
+     *         previous pairing
+     */
+    public boolean generateVerificationCode(String pairingCode) {
+        if (thingConfig.getPairingCode().equals(pairingCode) && !thingConfig.getVerificationCode().isEmpty()) {
+            logger.debug("Pairing code ({0}) refreshed for device {1} ({2}), verificationCode={3}", pairingCode,
+                    thingConfig.getFriendlyName(), thingConfig.getTerminalID(), thingConfig.getVerificationCode());
+            return false;
+        }
+        thingConfig.setPairingCode(pairingCode);
+        String md5Input = pairingCode + thingConfig.getTerminalID() + thingConfig.getUserID();
+        thingConfig.setVerificationCode(computeMD5(md5Input).toUpperCase());
+        logger.debug("verificationCode for device {0} ({1}): Input={2}, code={3}", thingConfig.getFriendlyName(),
+                thingConfig.getTerminalID(), md5Input, thingConfig.getVerificationCode());
+        return true;
+    }
+
+    /**
+     * Send a pairing verification request to the receiver. This is important to
+     * complete the pairing process. You should see a message like "Connected to
+     * openHAB" on your TV screen.
+     *
+     * @return true: successful, false: a non-critical error occured, caller handles
+     *         this
+     * @throws MagentaTVException
+     */
+    public boolean verifyPairing() throws MagentaTVException {
+        logger.debug("Verify pairing for device {0} (id={1}, code={2}", thingConfig.getFriendlyName(),
+                thingConfig.getTerminalID(), thingConfig.getVerificationCode());
+        String soapBody = MessageFormat.format(PAIRCHECK_SOAP_BODY, thingConfig.getTerminalID(),
+                thingConfig.getVerificationCode());
+        String soapXml = MessageFormat.format(SOAP_ENVELOPE, soapBody);
+        String response = http.httpPOST(buildHost(), buildReceiverUrl(PAIRCHECK_URI), soapXml, PAIRCHECK_SOAP_ACTION,
+                CONNECTION_CLOSE);
+
+        // Exception if request failed (response code != HTTP_OK)
+        if (!response.contains("<pairingResult>")) {
+            throw new MagentaTVException("Unexpected result for pairing verification: " + response);
+        }
+
+        String result = getXmlValue(response, "pairingResult");
+        if (!result.equals("0")) {
+            logger.info("Pairing failed or pairing no longer valid, result={0}", result);
```

Comment:
change to logger .debug

Reply:
important info, could indicate wrong credentials

Reply:
Could be added to Thing status.